core/services/ocr2/plugins/ocr2keeper/evm21/upkeep_state_store.go

```diff
@@ -0,0 +1,175 @@
+package evm
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"sync"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+
+	mapset "github.com/deckarep/golang-set/v2"
+)
+
+const Separator = "|"
+
+type upkeepState struct {
+	payload  *ocr2keepers.UpkeepPayload
+	state    *UpkeepState
+	block    int64
+	upkeepId string
+}
+
+type UpkeepState uint8
```

Comment:
We need this struct in `ocr2keepers` so it would be possible to call `SetUpkeepState` from there

Reply:
i added already but i don't think we cut a new release for it yet. hence copy it here for now

Reply:
Could you add a todo?

Reply:
done